### ISSUE TITLE
misalignment of graphic layer

### ISSUE THREAD

**Issue by femifrak** [jbarlow83/OCRmyPDF#49]
Sometimes the graphic layer is misaligned while the text layer seems to be placed correctly. I uploaded a sample pdf (test07.pdf) at:

http://www.loaditup.de/838186-ns8hr3kcbg.html

`ocrmypdf --oversample 600 test07.pdf test07ocr.pdf`

shows what I mean. test07ocr.pdf can be seen here:
http://www.loaditup.de/838187-4hkqhkbvnm.html

Additionally ocrmypdf gives a warning:
```
   **** File did not complete the page properly and may be damaged.

   **** This file had errors that were repaired or ignored.
   **** The file was produced by: 
   **** >>>> PyPDF2 <<<<
   **** Please notify the author of the software that produced this
   **** file that it does not conform to Adobe's published PDF
   **** specification.

```
I don't know whether this warning is justified. At least I have no problems in viewing the pdf in common pdf viewers. Have you got any idea about this?

**Comment by jbarlow83** [comment]
Thanks for the report. The image is visually centred in the page, but the page in this PDF is not located at (0, 0) in PDF coordinates, which is allowed but unusual. 

I should have a fix working for the next release.

The warning is for an intermediate single page PDF produced by ocrmypdf using PyPDF2, not your input PDF.

A temporary workaround which fixes both issues is to use `--force-ocr`. The drawback of using this is that the image will be transcoded, but with a high quality scan and --oversample set you will probably not notice any quality loss.

**Comment by femifrak** [comment]
Thanks!

Could you please provide the command on the main page how to update/uninstall ocrmypdf HEAD revision from sources?

Because of my weak memory i always
* search for the right command,
* perform it `(sudo pip3 uninstall ocrmypdf)`, and then
* follow your installation instructions.

(This is my new memo from now on.)

**Comment by jbarlow83** [comment]
Remember I haven't released the change yet, but in one shot you can upgrade without uninstall/reinstall with the following (provided pip is reasonably new, all ducks are sufficiently aligned, enough gods have been sacrificed too):
```
[sudo] pip3 install --upgrade git+https://github.com/jbarlow83/OCRmyPDF.git
```

That is buried in the main page, but I will update that to make it clear.

**Comment by jbarlow83** [comment]
Fixed in v4.0. 
Hopefully.
